nuxt-app/src/store/index.js

```diff
@@ -185,21 +186,42 @@ export const mutations = {
     state.vocab = vocabMap;
   },
   SET_VOCAB_CLASSES(state, vocabMap) {
-    // Set vocabClasses to an array of objects
+    // Set vocabClasses to a map (as in vue-client/src/store.js)
     const classTerms = [].concat(
       VocabUtil.getTermByType('Class', vocabMap, state.vocabContext, state.settings),
       VocabUtil.getTermByType('marc:CollectionClass', vocabMap, state.vocabContext, state.settings),
     );
-    state.vocabClasses = classTerms;
+    const classes = new Map(classTerms.map(entry => [entry['@id'], entry]));
+    classes.forEach((classObj) => {
+      if (classObj.hasOwnProperty('subClassOf')) {
+        each(classObj.subClassOf, (baseClass) => {
+          const baseClassObj = classes.get(baseClass['@id']);
+          if (typeof baseClassObj !== 'undefined') {
+            if (baseClassObj.hasOwnProperty('baseClassOf')) {
+              baseClassObj.baseClassOf.push(StringUtil.convertToPrefix(classObj['@id'], state.vocabContext));
+            } else {
+              baseClassObj.baseClassOf = [StringUtil.convertToPrefix(classObj['@id'], state.vocabContext)];
+            }
+          }
+        });
+      }
+    });
+    state.vocabClasses = classes;
   },
   SET_VOCAB_PROPERTIES(state, vocabMap) {
-    // Set vocabProperties to an array of objects
+    // Set vocabProperties to a map (as in vue-client/src/store.js)
     let props = [];
     props = props.concat(VocabUtil.getTermByType('Property', vocabMap, state.vocabContext, state.settings));
     props = props.concat(VocabUtil.getTermByType('DatatypeProperty', vocabMap, state.vocabContext, state.settings));
     props = props.concat(VocabUtil.getTermByType('ObjectProperty', vocabMap, state.vocabContext, state.settings));
     props = props.concat(VocabUtil.getTermByType('owl:SymmetricProperty', vocabMap, state.vocabContext, state.settings));
-    state.vocabProperties = props;
+    const vocabProperties = new Map(props.map(entry => [entry['@id'], entry]));
+
+    state.vocabClasses.forEach((classObj) => {
+      const compactClassUri = StringUtil.getCompactUri(classObj['@id'], state.vocabContext)
+      const allowedPropertiesInClass = VocabUtil.getProperties(compactClassUri, state.vocabClasses, vocabProperties, state.vocabContext)
```

Comment:
The returned value from `const allowedPropertiesInClass = VocabUtil.getProperties(...` isn't actually used here but `getProperties` adds `allowedProperties` to the classes in the vuex-store (due to mutation). I'm thinking I should restructure this in some way but need to discuss it...